channeldb/graph.go

```diff
@@ -1675,11 +1675,17 @@ func (c *ChannelGraph) FilterKnownChanIDs(chanIDs []uint64) ([]uint64, error) {
 }
 
 // FilterChannelRange returns the channel ID's of all known channels which were
-// mined in a block height within the passed range. This method can be used to
-// quickly share with a peer the set of channels we know of within a particular
-// range to catch them up after a period of time offline.
-func (c *ChannelGraph) FilterChannelRange(startHeight, endHeight uint32) ([]uint64, error) {
-	var chanIDs []uint64
+// mined in a block height within the passed range. The channel IDs are grouped
+// by their common block height. This method can be used to quickly share with a
+// peer the set of channels we know of within a particular range to catch them
+// up after a period of time offline.
+func (c *ChannelGraph) FilterChannelRange(startHeight,
+	endHeight uint32) ([][]lnwire.ShortChannelID, error) {
```

Comment:
Hard to know at a glance exactly what this new return type is at a glance, could instead use something like: 
```go
type ChanIDBlockRange struct {
    BlockHeight uint32
    ChanIDs []lnwire.ShortChannelID 
}
```
This may also simplify some logic elsewhere in the PR as well. 

We'd then return `[]ChanIDBlockRange` here.

Reply:
This new struct or even over `[] ChanIDBlockRange` could then grow to encapsulate the: "add chan ID to block height range" or "add new block height range" logic which is unrolled below.

Reply:
This would also let much of this logic be unit tested in isolation as well.

Reply:
Went with a hybrid approach of this and a map that's later sorted as suggested by @halseth.